Lib/ufo2ft/compilers/baseCompiler.py

```diff
@@ -0,0 +1,304 @@
+import logging
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Callable, Optional, Type
+
+from fontTools.designspaceLib.split import splitInterpolable, splitVariableFonts
+from fontTools.misc.loggingTools import Timer
+from fontTools.otlLib.optimize.gpos import GPOS_COMPACT_MODE_ENV_KEY
+
+from ufo2ft.constants import MTI_FEATURES_PREFIX
+from ufo2ft.errors import InvalidDesignSpaceData
+from ufo2ft.featureCompiler import FeatureCompiler, MtiFeatureCompiler
+from ufo2ft.postProcessor import PostProcessor
+from ufo2ft.util import (
+    _notdefGlyphFallback,
+    colrClipBoxQuantization,
+    ensure_all_sources_have_names,
+    location_to_string,
+    prune_unknown_kwargs,
+)
+
+
+@dataclass
+class BaseCompiler:
+    postProcessorClass: Type = PostProcessor
+    featureCompilerClass: Optional[Type] = None
+    featureWriters: list = None
+    filters: list = None
+    glyphOrder: list = None
+    useProductionNames: bool = None
+    removeOverlaps: bool = False
+    overlapsBackend: str = None
+    inplace: bool = False
+    layerName: str = None
+    skipExportGlyphs: bool = None
+    debugFeatureFile: str = None
+    notdefGlyph: str = None
+    colrLayerReuse: bool = True
+    colrAutoClipBoxes: bool = True
+    colrClipBoxQuantization: Callable[[object], int] = colrClipBoxQuantization
+    feaIncludeDir: str = None
+    skipFeatureCompilation: bool = False
+    flattenComponents: bool = None
```

Comment:
I think this one should only apply to truetype compilers, not to the BaseCompiler, no?